controllers/scan_test.go

```diff
@@ -311,5 +312,35 @@ var _ = Describe("ImageRepository controller", func() {
 			}, timeout, interval).Should(BeTrue())
 			Expect(ready.Message).To(ContainSubstring("should not start with URL scheme"))
 		})
+		It("does not fail if using a hostname with a port number", func() {
+			imgRepo := loadImages(registryServer, "test-fetch", []string{"1.0.0"})
+			imgRepo = strings.ReplaceAll(imgRepo, "127.0.0.1", "localhost")
```

Comment:
I didn't realize the behavior would change when dealing with hostnames vs an IP address. Here's a test to cover that situation.

Reply:
Is there a port in number in this test?

Reply:
I don't think it is -- https://play.golang.com/p/0Vw2HyPHb4P. Now I don't know how it worked before :-S

Reply:
@stefanprodan there is although not explicitly, the `registryServer` listens on a random port which ends up being included in the URI (e.g. `127.0.0.1:33571/test-fetch`).

@squaremo it does change, try with `localhost:5000/bar` which will succeed parsing but will fail to identify the (missing) scheme. The IP address format will just fail to parse. It worked before because we didn't look at the error but rather just checked if the parsing succeeded and if there was a `Scheme` attribute set (which was not correctly set in the situation when a port would be part of the URI when using a proper hostname).

Reply:
I see, `localhost:5000/bar` parses as a URL with scheme of `localhost` and a path, so falls foul of that condition; but `127.0.0.1:5000` doesn't parse as a URL, and skips ahead.